### ISSUE TITLE
Embeddings plugin example generates undefined property

### ISSUE THREAD

**Issue by tahpot** [oramasearch/orama#849]
### Describe the bug

I have copied the exact code from the embeddings example (https://docs.orama.com/open-source/plugins/plugin-embeddings) and I receive the following error when attempting to search:

```
node_modules/@orama/orama/dist/commonjs/methods/search-vector.js:17
    const vectorIndex = orama.data.index.vectorIndexes[vector.property];
                                                              ^

TypeError: Cannot read properties of undefined (reading 'property')
```

### To Reproduce

Execute the exact demo code here: https://docs.orama.com/open-source/plugins/plugin-embeddings

### Expected behavior

Expected a search result, but receive an internal error in Orama.

### Environment Info

```bash
OS: Sonoma 14.6.1
Arch: Apple M1 Max ARM64
Node: 18.20.5
Orama: 3.0.2
```

### Affected areas

Search

**Comment by micheleriva** [comment]
Looking into that right now

**Comment by micheleriva** [comment]
Issue identified. It requires a pretty big refactoring but I'll try to make it this week.

**Comment by micheleriva** [comment]
Fixed in #852, will make a release today
